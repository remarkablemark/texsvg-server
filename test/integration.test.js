const supertest = require('supertest');
const app = require('../app');
let { contentType, imageSvgXml, emptySvg } = require('../helpers/constants');
const svgs = require('../helpers/svgs');

imageSvgXml += '; charset=utf-8';
const status200 = 200;

const agent = supertest.agent(app);

describe('GET /heartbeat', () => {
  it('returns with 200 OK', async () => {
    await agent.get('/heartbeat').expect(status200, 'OK');
  });
});

describe('GET /', () => {
  it('returns with empty SVG', async () => {
    await agent
      .get('/')
      .expect(contentType, imageSvgXml)
      .expect(status200)
      .expect(res => {
        expect(res.body.toString()).toBe(emptySvg);
      });
  });
});

const tex1 = 'x';
describe(`GET /?tex=${tex1}`, () => {
  let svg;

  it('does not have SVG memoized', () => {
    expect(svgs[tex1]).toBe(undefined);
  });

  it('returns with SVG', async () => {
    await agent
      .get(`/?tex=${tex1}`)
      .expect(contentType, imageSvgXml)
      .expect(status200)
      .expect(res => {
        svg = res.body.toString();
        expect(svg).toMatchSnapshot();
      });
  });

  it('has SVG memoized', () => {
    expect(svgs[tex1]).toBe(svg);
  });
});

const quadraticFormula = 'x=\\frac{-b\\pm\\sqrt{b^2-4ac}}{2a}';
const tex2 = encodeURIComponent(quadraticFormula);
describe(`GET /?tex=${tex2}`, () => {
  let svg;

  it('does not have SVG memoized', () => {
    expect(svgs[quadraticFormula]).toBe(undefined);
  });

  it('returns with SVG', async () => {
    await agent
      .get(`/?tex=${tex2}`)
      .expect(contentType, imageSvgXml)
      .expect(status200)
      .expect(res => {
        svg = res.body.toString();
        expect(svg).toMatchSnapshot();
      });
  });

  it('has SVG memoized', () => {
    expect(svgs[quadraticFormula]).toBe(svg);
  });
});
