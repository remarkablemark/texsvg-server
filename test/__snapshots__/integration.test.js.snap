// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GET /?tex=x returns with SVG 1`] = `"<svg style=\\"vertical-align:-.025ex\\" xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"1.294ex\\" height=\\"1.025ex\\" viewBox=\\"0 -442 572 453\\" xmlns:xlink=\\"http://www.w3.org/1999/xlink\\"><defs><path id=\\"a\\" d=\\"M52 289q7 42 54 97t116 56q35 0 64-18t43-45q42 63 101 63 37 0 64-22t28-59q0-29-14-47t-27-22-23-4q-19 0-31 11t-12 29q0 46 50 63-11 13-40 13-13 0-19-2-38-16-56-66-60-221-60-258 0-28 16-40t35-12q37 0 73 33t49 81q3 10 6 11t16 2h4q15 0 15-8 0-1-2-11-16-57-62-101T333-11q-70 0-106 63-41-62-94-62h-6q-49 0-70 26T35 71q0 32 19 52t45 20q43 0 43-42 0-20-12-35t-23-20-13-5l-3-1q0-1 6-4t16-7 19-3q36 0 62 45 9 16 23 68t28 108 16 66q5 27 5 39 0 28-15 40t-34 12q-40 0-75-32t-49-82q-2-9-5-10t-16-2H58q-6 6-6 11z\\"/></defs><g data-mml-node=\\"math\\" stroke=\\"currentColor\\" fill=\\"currentColor\\" stroke-width=\\"0\\"><use xlink:href=\\"#a\\" data-mml-node=\\"mi\\" transform=\\"scale(1 -1)\\"/></g></svg>"`;

exports[`GET /?tex=x%3D%5Cfrac%7B-b%5Cpm%5Csqrt%7Bb%5E2-4ac%7D%7D%7B2a%7D returns with SVG 1`] = `"<svg style=\\"vertical-align:-1.575ex\\" xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"20.69ex\\" height=\\"5.291ex\\" viewBox=\\"0 -1642.5 9145 2338.5\\" xmlns:xlink=\\"http://www.w3.org/1999/xlink\\"><defs><path id=\\"a\\" d=\\"M52 289q7 42 54 97t116 56q35 0 64-18t43-45q42 63 101 63 37 0 64-22t28-59q0-29-14-47t-27-22-23-4q-19 0-31 11t-12 29q0 46 50 63-11 13-40 13-13 0-19-2-38-16-56-66-60-221-60-258 0-28 16-40t35-12q37 0 73 33t49 81q3 10 6 11t16 2h4q15 0 15-8 0-1-2-11-16-57-62-101T333-11q-70 0-106 63-41-62-94-62h-6q-49 0-70 26T35 71q0 32 19 52t45 20q43 0 43-42 0-20-12-35t-23-20-13-5l-3-1q0-1 6-4t16-7 19-3q36 0 62 45 9 16 23 68t28 108 16 66q5 27 5 39 0 28-15 40t-34 12q-40 0-75-32t-49-82q-2-9-5-10t-16-2H58q-6 6-6 11z\\"/><path id=\\"b\\" d=\\"M56 347q0 13 14 20h637q15-8 15-20 0-11-14-19l-318-1H72q-16 5-16 20zm0-194q0 15 16 20h636q14-10 14-20 0-13-15-20H70q-14 7-14 20z\\"/><path id=\\"c\\" d=\\"M84 237v13l14 20h581q15-8 15-20t-15-20H98q-14 7-14 20z\\"/><path id=\\"d\\" d=\\"M73 647q0 10 4 23t12 13q1 0 72 5t73 6q12 0 12-9t-34-143q-8-34-17-70t-15-54l-4-19q0-3 6 3 49 40 101 40 62 0 100-46t39-116q0-111-79-201T173-11q-50 0-91 38T40 150v9q0 21 8 58t49 197q50 197 50 209t-38 14H83q-3 0-7 3t-3 7zm263-322v6q0 74-61 74-17 0-35-8t-33-21-26-24-18-22l-6-8-21-86q-22-86-22-122 0-48 24-72 16-16 40-16 33 0 67 32 25 23 40 56t33 105q18 72 18 106z\\"/><path id=\\"e\\" d=\\"M56 320v13l14 20h299v149q0 149 2 153 5 11 17 11 14 0 17-12t4-58V353h298q15-8 15-20 0-13-15-20H409V40h298q15-8 15-20T707 0H70Q56 7 56 20t14 20h299v273H70q-14 7-14 20z\\"/><path id=\\"j\\" d=\\"M95 178q-6 0-14 8t-9 14 31 30 66 50 38 29q2 2 5 2h1q6 0 14-17t54-117q19-43 31-69l85-185q1 0 104 213t206 429 107 221q6 14 20 14 7 0 12-6t7-12v-6L620 293 385-193q-4-7-19-7-9 0-12 3-2 2-98 212l-96 210-16-11q-15-12-31-24t-18-12z\\"/><path id=\\"f\\" d=\\"M109 429q-27 0-43 18t-16 44q0 71 53 123t132 52q91 0 152-56t62-145q0-43-20-82t-48-68-80-74q-36-31-100-92l-59-56 76-1q157 0 167 5 7 2 24 89v3h40v-3q-1-3-13-91T421 3V0H50v31q0 7 6 15t30 35q29 32 50 56 9 10 34 37t34 37 29 33 28 34 23 30 21 32 15 29 13 32 7 30 3 33q0 63-34 109t-97 46q-33 0-58-17t-35-33-10-19q0-1 5-1 18 0 37-14t19-46q0-25-16-42t-45-18z\\"/><path id=\\"g\\" d=\\"M462 0q-18 3-129 3-116 0-134-3h-9v46h58q7 0 17 2t14 5 7 8q1 2 1 54v50H28v46l151 231q153 232 155 233 2 2 21 2h18l6-6V211h92v-46h-92v-51-48q0-7 6-12 8-7 57-8h29V0h-9zM293 211v334L74 212l109-1h110z\\"/><path id=\\"h\\" d=\\"M33 157q0 101 76 192t171 92q51 0 90-49 16 30 46 30 13 0 23-8t10-20q0-13-37-160T374 68q0-25 7-33t21-9q9 1 20 9 21 20 41 96 6 20 10 21 2 1 10 1h4q19 0 19-9 0-6-5-27t-20-54-32-50Q436 0 417-8q-8-2-24-2-34 0-57 15t-30 31l-6 15q-1 1-4-1l-4-4q-59-56-120-56-55 0-97 40T33 157zm318 171q0 6-5 22t-23 35-46 20q-35 0-67-31t-50-81q-29-79-41-164v-11q0-8-1-12 0-45 18-62t43-18q38 0 75 33t44 51q2 4 27 107t26 111z\\"/><path id=\\"i\\" d=\\"M34 159q0 109 86 196t186 87q56 0 88-24t33-63q0-29-19-49t-48-21q-19 0-30 10t-11 30 11 34 22 21 14 6h1q0 2-6 6t-21 8-34 4q-30 0-57-14-21-9-43-31-44-44-64-124t-21-116q0-46 26-69 22-24 58-24h4q112 0 185 85 9 10 12 10 4 0 13-9t10-14-9-15-29-28-45-30-64-25-80-11q-75 0-121 48T34 159z\\"/></defs><g data-mml-node=\\"math\\" stroke=\\"currentColor\\" fill=\\"currentColor\\" stroke-width=\\"0\\"><use xlink:href=\\"#a\\" data-mml-node=\\"mi\\" transform=\\"scale(1 -1)\\"/><use xlink:href=\\"#b\\" data-mml-node=\\"mo\\" transform=\\"matrix(1 0 0 -1 849.8 0)\\"/><g data-mml-node=\\"mfrac\\"><g data-mml-node=\\"mrow\\"><use xlink:href=\\"#c\\" data-mml-node=\\"mo\\" transform=\\"matrix(1 0 0 -1 2125.6 -676)\\"/><use xlink:href=\\"#d\\" data-mml-node=\\"mi\\" transform=\\"matrix(1 0 0 -1 2903.6 -676)\\"/><use xlink:href=\\"#e\\" data-mml-node=\\"mo\\" transform=\\"matrix(1 0 0 -1 3554.8 -676)\\"/><g data-mml-node=\\"msqrt\\"><g data-mml-node=\\"msup\\"><use xlink:href=\\"#d\\" data-mml-node=\\"mi\\" transform=\\"matrix(1 0 0 -1 5408 -676)\\"/><use xlink:href=\\"#f\\" data-mml-node=\\"mn\\" transform=\\"matrix(.707 0 0 -.707 5837 -965)\\"/></g><use xlink:href=\\"#c\\" data-mml-node=\\"mo\\" transform=\\"matrix(1 0 0 -1 6462.8 -676)\\"/><use xlink:href=\\"#g\\" data-mml-node=\\"mn\\" transform=\\"matrix(1 0 0 -1 7463 -676)\\"/><use xlink:href=\\"#h\\" data-mml-node=\\"mi\\" transform=\\"matrix(1 0 0 -1 7963 -676)\\"/><use xlink:href=\\"#i\\" data-mml-node=\\"mi\\" transform=\\"matrix(1 0 0 -1 8492 -676)\\"/><g data-mml-node=\\"mo\\" transform=\\"matrix(1 0 0 -1 4555 -782.5)\\"><use xlink:href=\\"#j\\"/></g><path stroke=\\"none\\" d=\\"M5408-1522.5h3517v-60H5408z\\"/></g></g><g data-mml-node=\\"mrow\\"><use xlink:href=\\"#f\\" data-mml-node=\\"mn\\" transform=\\"matrix(1 0 0 -1 5010.8 686)\\"/><use xlink:href=\\"#h\\" data-mml-node=\\"mi\\" transform=\\"matrix(1 0 0 -1 5510.8 686)\\"/></g><path stroke=\\"none\\" d=\\"M2025.6-220H9025v-60H2025.6z\\"/></g></g></svg>"`;
